xds/internal/xdsclient/filter_chain.go

```diff
@@ -587,33 +588,35 @@ func processNetworkFilters(filters []*v3listenerpb.Filter) (*FilterChain, error)
 				// TODO: Implement terminal filter logic, as per A36.
 				filterChain.HTTPFilters = filters
 				seenHCM = true
-				switch hcm.RouteSpecifier.(type) {
-				case *v3httppb.HttpConnectionManager_Rds:
-					if hcm.GetRds().GetConfigSource().GetAds() == nil {
-						return nil, fmt.Errorf("ConfigSource is not ADS: %+v", hcm)
+				if env.RBACSupport {
+					switch hcm.RouteSpecifier.(type) { // Guard this routing parsing...what does this trigger in filter chain and more downstream (no, nil is completly valid all the way...)? Also guard routeAndProcess, this will prevent the logic of a nil route configuration)
```

Comment:
? comment?

Are we supposed to be disabling all this parsing & validation too?

Reply:
Same reason I guarded it in client. Oh whoops sorry for checking this in, will delete.